### ISSUE TITLE
Feature: Support for Vercel's Custom Environments in target field

### ISSUE THREAD

**Issue by Dvinge** [vercel/terraform-provider-vercel#245]
## Description
The current implementation of `vercel_project_environment_variable` and `vercel_project_environment_variables` resources restricts the `target` field to only accept `production`, `preview`, or `development`. However, Vercel has recently released Custom Environments feature which allows for user-defined environment names.

## Current Behavior
The provider enforces a strict validation on the `target` field, only allowing:
- production
- preview
- development

## Expected Behavior
The `target` field should accept any string value to support Vercel's Custom Environments feature, while maintaining backwards compatibility with the existing environment names.

## Example Usage
```hcl
resource "vercel_project_environment_variable" "example" {
  project_id = vercel_project.example.id
  key        = "API_KEY"
  value      = "secret-value"
  target     = ["staging"]  # Currently not possible
}
```

Documentation for Custom Environments: https://vercel.com/docs/deployments/custom-environments
Vercel CLI using `target` for environments: https://vercel.com/docs/cli/deploy#target

**Comment by ecrousseau** [comment]
I would also like to see this implemented.

**Comment by dglsparsons** [comment]
Coming soon - there's work in progress for this already! 🙏

**Comment by dglsparsons** [comment]
This is released in v2.8.0 of the Terraform provider. [docs](https://registry.terraform.io/providers/vercel/vercel/latest/docs/resources/project_environment_variable#custom_environment_ids-1). Albeit as a separate `custom_environment_ids` field, rather than in the `target` field. 🙏. 

Please update your provider, give it a go and let me know if you find any issues!
